File: app/src/main/java/at/tomtasche/reader/nonfree/AnalyticsManager.java
package at.tomtasche.reader.nonfree;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.util.Log;

public class AnalyticsManager {

    private boolean enabled;

    public void initialize(Context context) {
        return;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public void report(String event) {
        report(event, null, null);
    }

    public void report(String event, String key1, Object value1, String key2, Object value2) {
        if (!enabled) {
            return;
        }

        Bundle bundle = new Bundle();
        if (key1 != null) {
            bundle.putString(key1, String.valueOf(value1));
        }
        if (key2 != null) {
            bundle.putString(key2, String.valueOf(value2));
        }

        Log.i("smn", event);
    }

    public void report(String event, String key, Object value) {
        report(event, key, value, null, null);
    }

    public void setCurrentScreen(Activity activity, String name) {
        if (!enabled) {
            return;
        }

        Bundle bundle = new Bundle();
        bundle.putString("screen_name", name);
        bundle.putString("screen_class", activity.getClass().getSimpleName());

        Log.i("smn", name);
    }
}

File: app/src/main/java/at/tomtasche/reader/nonfree/AnalyticsManager.kt
package at.tomtasche.reader.nonfree

import android.app.Activity
import android.content.Context
import android.util.Log

/**
 * Reporting has been local logging only since the firebase analytics integration was removed; the
 * call sites are kept so an analytics backend can be wired back in here.
 */
class AnalyticsManager {

    private var enabled = false

    fun initialize(context: Context) {
        // nothing to set up while reporting is local only
    }

    fun setEnabled(enabled: Boolean) {
        this.enabled = enabled
    }

    @JvmOverloads
    fun report(
        event: String,
        key1: String? = null,
        value1: Any? = null,
        key2: String? = null,
        value2: Any? = null,
    ) {
        if (!enabled) {
            return
        }

        Log.i(
            TAG,
            buildString {
                append(event)
                if (key1 != null) append(" $key1=$value1")
                if (key2 != null) append(" $key2=$value2")
            },
        )
    }

    fun setCurrentScreen(activity: Activity, name: String) {
        if (!enabled) {
            return
        }

        Log.i(TAG, "screen $name (${activity.javaClass.simpleName})")
    }

    private companion object {
        const val TAG = "smn"
    }
}
